src/Mod/Draft/draftgeoutils/general.py

```diff
@@ -59,11 +59,17 @@ def precision():
     return precisionInt  # return PARAMGRP.GetInt("precision", 6)
 
 
-def vec(edge):
-    """Return a vector from an edge or a Part.LineSegment."""
-    # if edge is not straight, you'll get strange results!
-    if isinstance(edge, Part.Shape):
-        return edge.Vertexes[-1].Point.sub(edge.Vertexes[0].Point)
+def vec(edge, use_orientation = False):
+    """Return a vector from an edge or a Part.LineSegment.
+
+    If use_orientation is True, it takes into account the edges orientation.
+    If edge is not straight, you'll get strange results!
+    """
+    if isinstance(edge, Part.Edge):
```

Comment:
I'd be very careful with modifying this test, because before it was checking for `Part.Shape`, and now it's testing for `Part.Edge`. The `vec` function is used in many places so I would be very concerned about changing the behavior of many tools at the same time; if you only tested with Arch Structure you may not notice if the behavior of other tools changes as well (say offsets, or Wall, Panels, etc.).

Reply:
Indeed there could be a problem. I changed the tests to avoid this risk.